sim/battle.ts

```diff
@@ -1552,6 +1552,43 @@ export class Battle {
 			this.add('bigerror', `You will auto-tie if the battle doesn't end in ${turnsLeftText} (on turn 1000).`);
 		}
 
+		// Gen 1 Endless Battle Clause triggers
+		if (this.gen <= 1) {
+			let onlyTransform = true;
+			let onlyGhostsWithStruggle = true;
+			let onlyFrozen = true;
+			for (const side of this.sides) {
+				for (const pokemon of side.pokemon) {
+					if (pokemon.fainted) continue;
+					if (pokemon.status !== 'frz' as ID) {
+						onlyFrozen = false;
+					}
+
+					if (!(pokemon.set.moves.map(toID).includes('transform' as ID) && pokemon.set.moves.length <= 1) ||
+            (this.dex.currentMod === 'gen1stadium' && pokemon.species.id === 'ditto')) {
+						// Ditto naturally cannot cause an endless battle in Stadium, according to research
```

Comment:
How does Ditto work in Stadium?

Reply:
I DMed you this on discord. see https://www.smogon.com/forums/threads/endless-battle-clause-enhancements-for-rby.3673455/#post-8739130